notebooks/samples/104 - Train, Test, Evaluate for Regression with Auto Imports Dataset.ipynb

```diff
@@ -0,0 +1,329 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 104 - Train, Test, Evaluate for Regression with Auto Imports Dataset\n",
+    "\n",
+    "This sample notebook is based on the Gallery \"Sample 6: Train, Test, Evaluate for Regression: Auto Imports Dataset\" for AzureML Studio. This experiment demonstrates how to build a regression model to predict the automobile's price. The process includes training, testing, and evaluating the model on the Autom Imports data set.\n",
+    "\n",
+    "This sample demonstrates the use of several members of the mmlspark library:\n",
+    "- TrainRegressor\n",
+    "- SummarizeData\n",
+    "- CleanMissingData\n",
+    "- ComputeStatistics\n",
+    "- FindBestModel"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First, import packages"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import mmlspark"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Declare the schema for the data that will be read in. Allow all fields to be nullable, so that missing values can be handled appropriately, such as replacing them with the mean or median value for that column."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "from pyspark.sql.types import LongType, StringType, DoubleType, StructType, StructField\n",
+    "\n",
+    "tableSchema = StructType([StructField(\"symboling\",         LongType(),   True),\n",
+    "                          StructField(\"normalized-losses\", DoubleType(), True),\n",
+    "                          StructField(\"make\",              StringType(), True),\n",
+    "                          StructField(\"fuel-type\",         StringType(), True),\n",
+    "                          StructField(\"aspiration\",        StringType(), True),\n",
+    "                          StructField(\"body-style\",        StringType(), True),\n",
+    "                          StructField(\"drive-wheels\",      StringType(), True),\n",
+    "                          StructField(\"engine-location\",   StringType(), True),\n",
+    "                          StructField(\"wheel-base\",        DoubleType(), True),\n",
+    "                          StructField(\"length\",            DoubleType(), True),\n",
+    "                          StructField(\"width\",             DoubleType(), True),\n",
+    "                          StructField(\"height\",            DoubleType(), True),\n",
+    "                          StructField(\"curb-weight\",       LongType(),   True),\n",
+    "                          StructField(\"engine-type\",       StringType(), True),\n",
```

Comment:
Minor: use a list comprenension:
```python
[StructField(column[0], column[1](), True) for column in (("symboling", LongType), ("normalized-losses", DoubleType))]
```

Reply:
I made this change, it got over written; it will be in the next update